Crypto101.org

```diff
@@ -2912,7 +2912,38 @@ http://www.cs.ucdavis.edu/~rogaway/papers/relates.pdf for further reading
 
 *** MD5
 
-TODO: Explain MD5
+MD5 is a hash function designed by Ronald Rivest in 1991 as an extension of MD4. This hash function outputs 128-bit digests. In 2004 research conducted by Xiaoyun Wang, Dengguo Feng, Xuejia Lai and Hongbo Yu showed that MD5 is vulnerable to collision attacks, therefore it is recommended not to use this hash function when storing sensitive information such as passwords.
```

Comment:
This is true, but in a way that might be confusing to the reader. It's true that MD5 is not good for storing passwords; but that has always been the case, and isn't a consequence of its collision-resistance properties. 

Perhaps it would be better to talk about how MD5 has repeatedly been weakened by cryptanalysis, finally resulting in the paper you quote that results in a full collision-resistance break. This meant that MD5 was no longer suitable for applications like digital signatures; one year later, researchers were able to produce a forged TLS certificate due to this weakness. MD5 might still be secure, but is not recommended, in other contexts that are less demanding -- for example, the HMAC-MD5 MAC algorithm.

There should probably also be a BiBTeX citation for the paper.